cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -445,6 +445,30 @@ static void VisitTwoArrayValuesInline(const ArrayData& arr0, const ArrayData& ar
                         arr0.length, std::move(visit_valid), std::move(visit_null));
 }
 
+template <typename Arg0Type, typename Arg1Type, typename Arg2Type,
+	 typename VisitFunc, typename NullFunc>
+static void VisitThreeArrayValuesInline(const ArrayData& arr0, const ArrayData& arr1,
+                                      const ArrayData& arr2,
+				      VisitFunc&& valid_func, NullFunc&& null_func) {
```

Comment:
Is this indent accepted by lint?

Could you remove "WIP" from the pull request title?